beetsplug/lyrics.py

```diff
@@ -419,15 +419,23 @@ def fetch(self, artist, title):
             return None
 
         for hit in json["response"]["hits"]:
-            # Genius uses zero-width characters to denote lowercase
-            # artist names.
-            hit_artist = hit["result"]["primary_artist"]["name"]. \
-                strip(u'\u200b').lower()
+            hit_artist = self._clean(hit["result"]["primary_artist"]["name"])
 
-            if hit_artist == artist.lower():
+            if hit_artist.lower() == artist.lower():
                 return self.lyrics_from_song_page(hit["result"]["url"])
 
-        self._log.debug(u'genius: no matching artist')
+        self._log.debug(u'Genius failed to find a matching artist for \'{0}\'',
+                        artist)
+
+    def _clean(self, artist):
+        """Cleans genius-isms to help artist matching"""
+        # Genius uses zero-width characters to denote lowercase artist names
+        artist = artist.strip(u'\u200b')
+
+        # Genius uses minus-hypen compared to beets hyphen
+        artist = artist.replace(u'\u002D', u'\u2010')
```

Comment:
Hmm… I think it might be a bit more effective to sanitize the beets side with the opposite transformation (or just clean *both* sides). Otherwise, we'll miss cases where the beets database actually has a hyphen, which agrees with Genius. (Data in beets doesn't *always* come from MB…)

Reply:
Having consistent results with https://github.com/beetbox/beets/pull/3635/commits/7220c91d2abc3c2bb565765ab512f20c11946c12
although I'd like to confirm it with some tests as well